File: clients/csharp/examples/collections-replace-one-nested.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Replace a document
    var filter = Builders<Document>.CollectionFilter.Eq(
      "metadata.language",
      "English"
    );
    var replacement = new Document()
    {
      { "title", "Hidden Shadows of the Past" },
      { "number_of_pages", 481 },
      {
        "genres",
        new[] { "Biography", "Graphic Novel", "Dystopian", "Drama" }
      },
      {
        "metadata",
        new Document
        {
          { "isbn", "978-1-905585-40-3" },
          { "language", "French" },
          { "edition", "Anniversary Edition" },
        }
      },
    };
    var result = await collection.ReplaceOneAsync(filter, replacement);

    Console.WriteLine(result.MatchedCount);
    Console.WriteLine(result.ModifiedCount);
  }
}

File: clients/java/examples/collections-replace-one-nested.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.results.CollectionUpdateResult;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Filter;
import com.datastax.astra.client.core.query.Filters;
import java.util.Map;
import java.util.Set;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Replace a document
    Filter filter = Filters.eq("metadata.language", "English");
    Document newDocument =
        new Document()
            .append("title", "Hidden Shadows of the Past")
            .append("number_of_pages", 481)
            .append("genres", Set.of("Biography", "Graphic Novel", "Dystopian", "Drama"))
            .append(
                "metadata",
                Map.of(
                    "isbn", "978-1-905585-40-3",
                    "language", "French",
                    "edition", "Anniversary Edition"));
    CollectionUpdateResult result = collection.replaceOne(filter, newDocument);
    System.out.println(result.getMatchedCount());
    System.out.println(result.getModifiedCount());
  }
}
